app/pathogen/arbovirus/analyze/recharts/change-in-median-seroprevalence-over-time-graph.tsx

```diff
@@ -0,0 +1,196 @@
+    [0]?.groupingTimeInterval.intervalStartDate;
+
+  const latestDate = input.dataPoints
+    .toSorted((a, b) => isAfter(a.groupingTimeInterval.intervalEndDate, b.groupingTimeInterval.intervalEndDate) ? -1 : 1)
+    [0]?.groupingTimeInterval.intervalEndDate;
+
+  if (!earliestDate || !latestDate) {
+    return {
+      groupedDataPoints: [],
+    };
+  }
+
+  const { buckets } = generateBucketsForTimeInterval({
+    earliestDate,
+    latestDate,
+    groupingBucketSize: input.groupingBucketSize
+  })
+
+  let groupedDataPoints = buckets.map((bucket) => ({
+    interval: bucket,
+    dataPoints: input.dataPoints.filter((dataPoint) => doTimeIntervalsOverlap(bucket, dataPoint.groupingTimeInterval))
+  })).filter((bucketAndDataPoints) => bucketAndDataPoints.dataPoints.length > 0)
+
+  return {
+    groupedDataPoints,
+  };
+};
+
+export const ChangeInMedianSeroprevalenceOverTimeGraph = (): React.ReactNode => {
+  const state = useContext(ArboContext);
+
+  const dataGroupedByArbovirus = typedGroupBy(
+    state.filteredData,
+    (dataPoint) => dataPoint.pathogen
+  );
+
+  const dataGroupedByArbovirusThenByTimeBucket = typedObjectFromEntries(
+    typedObjectEntries(dataGroupedByArbovirus).map(
+      ([arbovirus, dataPointsForAParticularArbovirus]) => [
+        arbovirus,
+        groupDataPointsIntoTimeBuckets({
+          dataPoints: dataPointsForAParticularArbovirus.map((dataPoint) => ({
+            ...dataPoint,
+            groupingTimeInterval: {
+              intervalStartDate: parseISO(dataPoint.sampleStartDate),
+              intervalEndDate: parseISO(dataPoint.sampleEndDate),
+            },
+          })),
+          groupingBucketSize: {
+            years: 5,
+          },
+        }).groupedDataPoints,
+      ]
+    )
+  );
+
+  return (
+    <div className="h-full flex flex-col">
+      <div className="h-full flex flex-row flex-wrap">
+        {Object.keys(dataGroupedByArbovirusThenByTimeBucket).map((arbovirus, index) => {
+          const dataForArbovirus = dataGroupedByArbovirusThenByTimeBucket[arbovirus].map(({interval, dataPoints}) => ({
+            intervalAsString: `${interval.intervalStartDate.getFullYear()}-${interval.intervalEndDate.getFullYear()}`,
+            dataPoints,
+            median: median(dataPoints.map((dataPoint) => dataPoint.seroprevalence * 100))
+          }));
+
+          const numberOfSubgraphsDisplayed = Object.keys(dataGroupedByArbovirus).length;
+
+          const width = numberOfSubgraphsDisplayed < 3 ? "w-full" : "w-1/2";
+          const height =
+            numberOfSubgraphsDisplayed === 1
+              ? "h-full"
+              : numberOfSubgraphsDisplayed < 5
+              ? "h-1/2"
+              : "h-1/3";
+
+          return (
+            <div
+              className={clsx(width, height)}
+              key={`change-in-med-sero-prev-${arbovirus}`}
+            >
+              <h2 className="w-full text-center ">
+                {convertArboSFtoArbo(arbovirus as arbovirusesSF)}
```

Comment:
Sorry, I'm hoping to use a graphql typegen library on this repo eventually that should hopefully let us get better types and not have to do this.